File: Spring/src/main/java/com/grupoMarianaOttony/ApiControleFinanceiro/enums/Tipo.java
package com.grupoMarianaOttony.ApiControleFinanceiro.enums;

import com.fasterxml.jackson.annotation.JsonValue;

public enum Tipo {
    RECEITA("Receita"),
    DESPESA("Despesa");

    private final String value;

    Tipo(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return value;
    }
}

File: Angular/api-controle-financeiro/src/app/model/tipo.ts
export enum Tipo{
    Receita = 'Receita',
    Despesa = 'Despesa'
}

// optional: Record type annotation guaranties that 
// all the values from the enum are presented in the mapping
export const TipoLabelMapping: Record<Tipo, string> = {
    [Tipo.Receita]: "Receita",
    [Tipo.Despesa]: "Despesa",
};